sklearn/metrics/tests/test_score_objects.py

```diff
@@ -677,6 +681,7 @@ def test_regression_scorer_sample_weight():
             # skip classification scorers
             continue
         try:
+            scorer = scorer.set_score_request(sample_weight=True)
             weighted = scorer(reg, X_test, y_test, sample_weight=sample_weight)
             ignored = scorer(reg, X_test[11:], y_test[11:])
             unweighted = scorer(reg, X_test, y_test)
```

Comment:
Shouldn't those two raise because `sample_weight` was requested but not passed?

Reply:
validation only happens on the *router* level. This added line here is actually not needed anyway.